src/example-types/tagsQuery.js

```diff
@@ -0,0 +1,89 @@
+let _ = require('lodash/fp')
+let { parens } = require('futil-js')
+let Combinatorics = require('js-combinatorics')
+
+let wordPermutations = word =>
+  _.map(_.join(' '), Combinatorics.permutation(word.split(' ')).toArray())
+
+let limitResultsToCertainTags = tags => !!_.find('onlyShowTheseResults', tags)
+
+let wrapIf = _.curry(
+  (pre, post, text, shouldWrap) => (shouldWrap ? `${pre}${text}${post}` : text)
+)
+let quoteIf = wrapIf('"', '"')
+
+let quoteAndTilde = _.curry(
+  (tag, text) =>
+    quoteIf(text, tag.isPhrase) +
```

Comment:
`isPhrase` is just defined as the word string having a space in it, but I don't see that logic here. Are we expecting the client to send it up? For what it's worth, there's probably no harm at all in just always wrapping each tag in quotes, regardless of `isPhrase` or not. That might make some of this easier to follow (especially since you could remove the `quoteIf` function and just use `F.quote`

Reply:
`isPhrase` is handled by the client: https://github.com/smartprocure/main-application/blob/52c3e09844b287d362bebda5afe05b26b5ed7807/client/public/DataContext/Types/queryFilterNew.js#L52-L57. FYI: `F.quote` is undocumented. Good to know about that, though.